mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -621,29 +674,80 @@ def _compare_metrics(metric_a, metric_b):
                 metric_b.value,
             )
 
-        # Fetch the latest metric value corresponding to the specified run_id and metric key and
-        # lock its associated row for the remainder of the transaction in order to ensure
+        def _merge_metric(new_metric, old_metric):
```

Comment:
Nit: I think `_overwrite_metric` is a more descriptive name here

```suggestion
        def _overwrite_metric(new_metric, old_metric):
```

Reply:
Applied your suggestion, that was my other alternative :)